pkg/state/state_nic_feature_discovery.go

```diff
@@ -129,7 +129,7 @@ func (s *stateNICFeatureDiscovery) Sync(
 	return syncState, nil
 }
 
-// Get a map of source kinds that should be watched for the state keyed by the source kind name
+// GetWatchSources returns a map of source kinds that should be watched for the state keyed by the source kind name
```

Comment:
lint did not catch this before?

Reply:
No - there's a bunch of places where it's not catching issues. I'm trying to figure out what the root cause is.

Reply:
The issue seems to be that the linter is having trouble catching issues on implementation of methods from the interface. Not sure what the root issue is - but my IDE is able to detect those missing comments.